### PR TITLE
OPT (#54): Removed id as param in endpoint

### PR DESCRIPTION
**Pull request by tien-chung**
- Removed id from getRecipe()
- Created another endpoint function with the name of getRecipeId
- Updated RecipeControllerTest to have 100% coverage

Screenshot of testing param for id:
Screenshot 2024-11-28 at 1 29 33 PM

Screenshot of testing for owner:
Screenshot 2024-11-28 at 1 30 27 PM

Screenshot of testing both owner and id that should not work:
Screenshot 2024-11-28 at 1 30 52 PM

Screenshot of 100% testing coverage:
Screenshot 2024-11-28 at 1 31 38 PM

### CODE REVIEW

**Comment by ThiiLuu79** [comment]
Can you also update the README?

**Comment by ThiiLuu79** [comment]
Can you also update the description in the README. Some of them does not apply anymore.

**Comment by tien-chung** [comment]
> Can you also update the description in the README. Some of them does not apply anymore.

Fixed. My bad, I didn't see you have an example for id before.

File: README.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ The application should now be running on `http://localhost:8080`.
 **Method:** `GET`
 
 **Description:** This endpoint retrieves personal recipes. If the `owner` parameter is provided, it filters recipes by the specified owner. If the `owner` parameter is not provided, it returns all recipes.   
-If the `id` parameter is provided, it filters recipes by the specified id. If the `id` parameter is not provided, it returns all recipes.   
+The `id` path is provided after `persona/{id}`, it filters recipes by the specified id. If the `id` path is not provided, it returns all recipes.
```
Comment (ThiiLuu79) [inline README.md]: please change persona for personal

**Review by ThiiLuu79 — APPROVED** [review]

**Pull request merged**